src/viadot/sources/business_core.py

```diff
@@ -0,0 +1,177 @@
+    """Class to connect to Business Core ERP API."""
+
+    def __init__(
+        self,
+        url: str | None = None,
+        filters_dict: dict[str, Any] | None = None,
+        credentials: dict[str, Any] | None = None,
+        config_key: str = "BusinessCore",
+        verify: bool = True,
+        *args,
+        **kwargs,
+    ):
+        """Creating an instance of BusinessCore source class.
+
+        Args:
+            url (str, optional): Base url to a view in Business Core API.
+                Defaults to None.
+            filters_dict (Dict[str, Any], optional): Filters in form of dictionary.
+                Available filters: 'BucketCount', 'BucketNo', 'FromDate', 'ToDate'.
+                Defaults to None.
+            credentials (Dict[str, Any], optional): Credentials stored in a dictionary.
+                Required credentials: username, password. Defaults to None.
+            config_key (str, optional): Credential key to the dictionary where details
+                are stored. Defaults to "BusinessCore".
+            verify (bool, optional): Whether or not verify certificates while
+                connecting to an API. Defaults to True.
+
+        Raises:
+            CredentialError: When credentials are not found.
+        """
+        raw_creds = credentials or get_source_credentials(config_key) or None
+        error_message = "Missing Credentials."
+        if raw_creds is None:
+            raise CredentialError(error_message)
+
+        validated_creds = dict(BusinessCoreCredentials(**raw_creds))
+
+        self.credentials = validated_creds
+        self.url = url
+        self.filters_dict = filters_dict
+        self.verify = verify
+
+        super().__init__(*args, credentials=self.credentials, **kwargs)
+
+    def generate_token(self) -> str:
+        """Function for generating Business Core token based on username and password.
+
+        Returns:
+            string: Business Core API token.
+        """
+        url = "https://api.businesscore.ae/api/user/Login"
+
+        payload = f'grant_type=password&username={self.credentials.get("username")}&password={self.credentials.get("password")}&scope='
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        response = handle_api_response(
+            url=url,
+            headers=headers,
+            method="GET",
+            data=payload,
+            verify=self.verify,
+        )
+        token = json.loads(response.text).get("access_token")
+        self.token = token
+        return token
+
+    def clean_filters_dict(self) -> dict:
+        """Function for replacing 'None' with '&' in a dictionary.
+
+            Needed for payload in 'x-www-form-urlencoded' from.
+
+        Returns:
+            Dict: Dictionary with filters prepared for further use.
+        """
+        return {
+            key: ("&" if val is None else val) for key, val in self.filters_dict.items()
+        }
+
+    def get_data(self) -> dict:
```

Comment:
Should be refactored into `to_records()`; see existing issues regarding `api_connection()`, as this is the same case here. For now pls create an issue and link it here.

Reply:
created issue: https://github.com/dyvenia/viadot/issues/1078